zzre/game/systems/ui/ScrBookMenu.cs

```diff
@@ -42,27 +69,111 @@ protected override void HandleOpen(in messages.ui.OpenBookMenu message)
+                    preload.CreateImage(entity)
+                        .With(Mid + FairyButtonPos(i))
+                        .With(preload.Inf000, 16)
+                        .WithRenderOrder(-1)
+                        .Build();
+                }
             }
         }
     }
 
+    private DefaultEcs.Entity CreateSidebar(UIPreloader preload, in DefaultEcs.Entity parent, FairyRow fairyRow, ref components.ui.ScrBookMenu book)
+    {
+        var entity = World.CreateEntity();
+        entity.Set(new components.Parent(parent));
+
+        var fairyI = Array.FindIndex(book.Fairies, row => row.Name == fairyRow.Name) + 1;
+
+        var element = new components.ui.ElementId(0);
+        preload.CreateButton(entity)
+            .With(element)
+            .With(Mid + new Vector2(160, 218))
+            .With(new components.ui.ButtonTiles(fairyRow.CardId.EntityId))
+            .With(preload.Wiz000)
+            .Build();
+
+        preload.CreateLabel(entity)
+            .With(Mid + new Vector2(21, 57))
+            .WithText($"#{fairyI} {fairyRow.Name}")
+            .With(preload.Fnt000)
+            .Build();
+
+        preload.CreateImage(entity)
+            .With(Mid + new Vector2(22, 81))
+            .With(preload.Cls000, (int)fairyRow.Class0)
+            .Build();
+
+        preload.CreateLabel(entity)
+            .With(Mid + new Vector2(36, 80))
+            .WithText(db.GetText(UIDClassNames[(int)fairyRow.Class0-1]).Text)
+            .With(preload.Fnt002)
+            .Build();
+
+        if (fairyRow.EvolVar != -1)
+            preload.CreateLabel(entity)
+                .With(Mid + new Vector2(22, 246))
+                .WithText($"Evolution at level {fairyRow.EvolVar}")
+                .With(preload.Fnt002)
+                .Build();
+
+        CreateStat(preload, entity, 0, Math.Min(500, fairyRow.MHP) / 100);
+        CreateStat(preload, entity, 1, fairyRow.MovSpeed + 1);
+        CreateStat(preload, entity, 2, fairyRow.JumpPower + 1);
+        CreateStat(preload, entity, 3, fairyRow.CriticalHit + 1);
+
+        var MaxTextWidth = 190f;
+        preload.CreateLabel(entity)
+            .With(Mid + new Vector2(21, 271+17*3+24))
+            .WithText(fairyRow.Info)
+            .With(preload.Fnt002)
+            .WithLineWrap(MaxTextWidth)
+            .Build();
+
+        return entity;
+    }
+
+    private void CreateStat(UIPreloader preload, in DefaultEcs.Entity entity, int index, int value)
+    {
+        preload.CreateLabel(entity)
+            .With(Mid + new Vector2(21, 271 + index*17))
+            .WithText(db.GetText(UIDStatNames[index]).Text)
+            .With(preload.Fnt002)
+            .Build();
+
+        preload.CreateLabel(entity)
+            .With(Mid + new Vector2(111, 266 + index*17))
+            .WithText(string.Concat(Enumerable.Repeat("{1017}", value)) + string.Concat(Enumerable.Repeat("{1018}", 5-value)))
```

Comment:
There is a [`String(char ch, int count)`](https://learn.microsoft.com/en-us/dotnet/api/system.string.-ctor?view=net-8.0#system-string-ctor(system-char-system-int32):~:text=and%20other%20versions-,String(Char%2C%20Int32),-Initializes%20a%20new) constructor you can use here